packages/cbioportal-utils/src/mutation/ProteinChangeUtils.ts

```diff
@@ -12,46 +14,48 @@ const PROTEIN_CHANGE_MATCHERS = [
  *
  * @return  protein location
  */
-export function getProteinPositionFromProteinChange(
-    proteinChange?: string
-): { start: number; end: number } | undefined {
-    if (!proteinChange) {
-        return undefined;
-    }
+export const getProteinPositionFromProteinChange = memoize({
```

Comment:
@adamabeshouse just curious: why does oncoprint use `getProteinPositionFromProteinChange`? Is it for annotation of OncoKB or something?

Reply:
feels slightly funky to me to `memoize` this function but not fix the calling logic (i.e. why would this be called multiple times for the same protein changes)? Seems to indicate it might be calling other functions multiple times too for the same mutation? Unless all the speed up comes from mutations that have different genes but the same protein change (this is prolly not the case, because I believe you were testing on a single gene query?)

Reply:
@inodb check this out https://github.com/cBioPortal/cbioportal-frontend/blob/master/src/pages/resultsView/ResultsViewPageStore.ts#L5215

This function is used to get the protein position of a COSMIC mutation record, for comparison with a given mutation for annotating. the COSMIC mutation records are grouped by keyword, so a mutation is checked based on its keyword and then the COSMIC records are looped through. So this would happen repeatedly for mutations with the same keyword (which, in the case of truncating, are not that descriptive).

This discussion has just made me realize how inefficient this process actually is, though. Maybe there is an algorithmic optimization to make instead of memoizing this.